### PR TITLE
Support for version number validation against mixins

### PR DESCRIPTION
**Pull request by dgannon991**
# What does this change
Added a check on version numbers if specified in your porter YAML

# What issue does it fix
Delivers towards #3136 

# Notes for the reviewer
@kichristensen - this is just a draft to make sure I'd got the right idea, before adding the checks into the linting side. I was fairly sure I'd understood, but thought it was worth double checking as it's been a little while :)

### CODE REVIEW

**Comment by kichristensen** [comment]
@dgannon991 Yes the format looks like what I thought about. Didn't really think about adding the full semver constraint, but I think it makes sense, and it would allow to later implement logic that could automatically download the right version if needed

**Comment by dgannon991** [comment]
Nice. I’ll try and get the other half of it down for Thursday. Cheers!

File: pkg/manifest/manifest.go [hunk 1]
```diff
@@ -676,7 +711,12 @@ func (m *MixinDeclaration) UnmarshalYAML(unmarshal func(interface{}) error) erro
 	}
 
 	for mixinName, config := range mixinWithConfig {
-		m.Name = mixinName
+		name, version, err := extractVersionFromName(mixinName)
```
Comment (kichristensen) [inline pkg/manifest/manifest.go:714]: Won't the name and version already have been set at line 690? Is it needed to also set it here?

**Review by kichristensen — COMMENTED** [review]
Looks really good, added a few small comments

File: pkg/manifest/manifest.go (hunk 1 above)
Comment (dgannon991) [reply pkg/manifest/manifest.go:714]: Ah sadly yes it is, there are two different logic branches (I think!) One for if you just specify a mixin name, like:
```yaml
mixins:
  - exec
  - terraform
```

And another if you provide config for the mixin, like:
```yaml
mixins:
  - terraform:
    clientVersion: 1.3.0
```

**Review by kichristensen — APPROVED** [review]

**Pull request merged**